app/serializers/current_user_serializer.rb

```diff
@@ -237,4 +238,8 @@ def second_factor_enabled
   def featured_topic
     object.user_profile.featured_topic
   end
+
+  def do_not_disturb_until
+    object.active_do_not_disturb_timings&.first&.ends_at
```

Comment:
```suggestion
    object.active_do_not_disturb_timings.maximum(:ends_at)
```
The relationship should always exist, and you needn't load all fields from the database if you only want one value, the latest date.